### DISCUSSION TITLE
Rule Proposal: No default values for @Optional parameters

### DISCUSSION THREAD

**Discussion by sv3k** [angular-eslint/angular-eslint#2021] [New Rule Proposals]
If you have an `@Optional()` parameter in a constructor, and it doesn't have a value provided - Angular will inject it as `null`. If some default value is needed for that optional parameter - there's a temptation to use default parameters language feature, like so:

```typescript
constructor(@Optional() myParam = myDefaultValue) {}
```

Unfortunately that won't work because default parameters are used only if no argument is provided or its value is `undefined`, so in the example you'll have `null` instead of the anticipated `myDefaultValue`.

Angular has a feature request to solve this issue, created years ago: https://github.com/angular/angular/issues/25395

Until then, it would be great to have a linter rule preventing the above scenario.

**Comment by JamesHenry** [comment]
This is no longer as relevant with the `inject()` function for DI, which is the recommended approach: https://github.com/angular/angular/issues/25395#issuecomment-2320964696
